components/esp_modem/examples/simple_cmux_client/components/SIM7070_gnss/SIM7070_gnss.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-FileCopyrightText: 2021-2022 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+//
+// Created on: 23.08.2022
+// Author: franz
+
+
+#pragma once
+
+
+#include "cxx_include/esp_modem_dce_factory.hpp"
+#include "cxx_include/esp_modem_dce_module.hpp"
+#include "nmea_parser.h"
+
+/**
+ * @brief Definition of a custom modem which inherits from the GenericModule, uses all its methods
+ * and could override any of them. Here, for demonstration purposes only, we redefine just `get_module_name()`
+ */
+class SIM7070_gnss: public esp_modem::SIM7070 {
```

Comment:
Could these two classes be merged?

Reply:
They were one class originally, but I specifically asked to make two classes, since the `GNSS` is a specialization of the `7070`. Alternatively we could "compose" the `SIM7070_gnss` somehow statically to avoid the inheritance here (but I personally find the inheritance more straight forward)

Reply:
I suggest adding a description on both classes and why they are needed. Also, the description above `SIM7070_gnss` is correct?

Reply:
Thanks, that was a copy paste.

Updated the description of both classes.